addons/modules/functions/fnc_projectiles_unit.sqf

```diff
@@ -0,0 +1,174 @@
+	private _height = _projectilePosASL # 2 - _targetPos # 2;
+	private _g = 9.8066;
+	//private _speed = 10 + _distance / 2 - _height / 2;
+	private _speed = getNumber (configFile >> "CfgMagazines" >> _magazine >> "initSpeed");
+	
+	private _angle = (acos((_g * _distance^2/_speed^2-_height)/(_projectilePosASL distance _targetPos)) + atan (_distance / _height)) / 2;
+	
+	if !(_angle isEqualType 0) then {
+		/* boost speed
+		systemChat format ["d:%1, h:%2, s:%3, a:%4",_distance, _height, _speed, _angle];
+		while {_speed < 20 && {!(_angle isEqualType 0)}} do {
+			_speed = _speed + 1;
+			_angle = (acos((_g * _distance^2/_speed^2-_height)/(_projectilePosASL distance _targetPos)) + atan (_distance / _height)) / 2;
+		};
+		_speed = _speed + 2;
+		_angle = (acos((_g * _distance^2/_speed^2-_height)/(_projectilePosASL distance _targetPos)) + atan (_distance / _height)) / 2;
+		*/
+		_angle = 45;
+	};
+	
+	if (_angle < 0) then {
+		_angle = _angle + 90;
+	};
+	
+	if (_angle > 45) then {
+		//_angle = 90 - _angle;
+	};
+	//systemChat format ["IA:%1",_angle];
+	
+	private _speedY = _speed * sin _angle;
+	private _speedx = _speed * cos _angle;
+	
+	private _vectorLOS = _projectilePosASL vectorFromTo _targetPos;
+	private _vectorDir = [_projectilePosASL # 0,_projectilePosASL # 1, 0] vectorFromTo [_targetPos # 0, _targetPos # 1, 0];
+	private _vectorLaunch = vectorNormalized (_vectorDir vectorAdd [0,0,_speedY/_speedX]);
+	
+	// check if using high angle
+	if _throwFlatTrajectory then {
+		private _angleLOS_Vert = acos (_vectorLOS vectorCos [0,0,1]);
+		private _angleHORZ_LOS = acos (_vectorDir vectorCos _vectorLOS);
+		private _angleLOS_Launch = acos (_vectorLOS vectorCos _vectorLaunch);
+		private _angleLaunch_Vert = acos (_vectorLaunch vectorCos [0,0,1]);
+		//systemChat format ["LV:%1, HL:%2, LA:%3, AV:%4",_angleLOS_Vert, _angleHORZ_LOS, _angleLOS_Launch, _angleLaunch_Vert];
+		
+		if (_angleLOS_Launch > (_angleLOS_Vert / 2)) then {
+			if (_angleLOS_Vert > 90) then {
+				_angleHORZ_LOS = -1 * _angleHORZ_LOS;
+			};
+			_angle = _angleLaunch_Vert + _angleHORZ_LOS;
+			//systemChat format ["FA:%1",_angle];
+			
+			_speedY = _speed * sin _angle;
+			_speedx = _speed * cos _angle;
+			_vectorLaunch = vectorNormalized (_vectorDir vectorAdd [0,0,_speedY/_speedX]);
+		};
+	};
+	
+	private _vectorFinal = _vectorLaunch vectorMultiply _speed;
+	// projectile orientation
+	private _vectorSide = _vectorFinal vectorCrossProduct [0,0,-1];
+	private _vectorUp = _vectorFinal vectorCrossProduct _vectorSide;
+	
+	_projectile setVectorDirAndUp [
+		_vectorFinal,
+		_vectorUp
+	];
+	
+	// 
+	_projectile setVelocity _vectorFinal;
+	
+	// clean up
+	_unit removeEventHandler ["Fired", _thisEventHandler];
+	_unit setVariable ["zen_projectiles_thrown", true];
+	_unit enableAI "PATH";
+	
+}];
+
+_unit setVariable ["zen_projectiles_thrown", false];
+_unit setVariable ["zen_projectiles_time", diag_tickTime];
+
+_this spawn {
```

Comment:
Use of spawn and waitUntil, could this not be achieved with CBA instead?

Reply:
New implementation with CBA.